media_service/mediastore.py

```diff
@@ -44,7 +45,7 @@ def processFileUploads(filelist):
                 newfile.name = f
 
                 # avoiding temp files added to archive
-                if "__MACOSX" not in newfile.name:
+                if "__MACOSX" not in newfile.name and not re.match(r".*\/$", newfile.name):
```

Comment:
@jazahn That regex does the job, but I'm thinking it might be a little clearer to push the filtering logic to the top of the inner loop like this:

```python
zip = zipfile.ZipFile(file, "r")
for f in zip.namelist():
    logger.debug("Extracting ZIP file: %s" % f)

    # Filter out unwanted files from the ZIP
    if f.endswith('/'):
        logger.debug("Skipping directory entry: %s" % f)
        continue
    if "__MACOSX" in f or ".DS_Store" in f:
        logger.debug("Skipping MAC OS X resource file: %s" % f)
        continue

    # Read the ZIP file to memory
    zf = zip.open(f).read()
    newfile = File(io.BytesIO(zf))
    newfile.name = f

    # Add the file to the upload list
    newlist.append(newfile)
```

A benefit of that method is it won't read unwanted files into memory, it will just skip them completely.

But I'm wondering if maybe we really should be filtering based on our desired file extensions (i.e. whitelisting) rather than trying to filter out unwanted files (blacklisting), since that would have avoided the directory issue from the beginning and would prevent other unwanted files from progressing to the validation stage. I'm not sure, though. What do you think?